osquery/filesystem/filesystem.cpp

```diff
@@ -51,49 +53,50 @@ Status writeTextFile(const fs::path& path,
                      int permissions,
                      bool force_permissions) {
   // Open the file with the request permissions.
-  int output_fd =
-      open(path.c_str(), O_CREAT | O_APPEND | O_WRONLY, permissions);
-  if (output_fd <= 0) {
+  PlatformFile output_fd(path.string(), PF_CREATE_ALWAYS | PF_WRITE, permissions);
+  if (!output_fd.isValid()) {
     return Status(1, "Could not create file: " + path.string());
   }
 
   // If the file existed with different permissions before our open
   // they must be restricted.
-  if (chmod(path.c_str(), permissions) != 0) {
+  if (!platformChmod(path.string(), permissions)) {
     // Could not change the file to the requested permissions.
     return Status(1, "Failed to change permissions for file: " + path.string());
   }
 
-  ssize_t bytes = write(output_fd, content.c_str(), content.size());
+  ssize_t bytes = output_fd.write(content.c_str(), content.size());
   if (static_cast<size_t>(bytes) != content.size()) {
-    close(output_fd);
     return Status(1, "Failed to write contents to file: " + path.string());
   }
 
-  close(output_fd);
   return Status(0, "OK");
 }
 
 struct OpenReadableFile {
- public:
-  explicit OpenReadableFile(const fs::path& path) {
+public:
+  explicit OpenReadableFile(const fs::path &path) {
+#ifndef WIN32
     dropper_ = DropPrivileges::get();
     if (dropper_->dropToParent(path)) {
       // Open the file descriptor and allow caller to perform error checking.
-      fd = open(path.string().c_str(), O_RDONLY | O_NONBLOCK);
+      fd.reset(new PlatformFile(path.string().c_str(),
```

Comment:
Why use `.c_str()` here? Also why duplicate this call here and within the `#else`?